server/src/main/java/org/jetbrains/bsp/bazel/server/bloop/BloopExporter.java

```diff
@@ -0,0 +1,166 @@
+package org.jetbrains.bsp.bazel.server.bloop;
+
+import ch.epfl.scala.bsp4j.BuildClient;
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import ch.epfl.scala.bsp4j.DiagnosticSeverity;
+import ch.epfl.scala.bsp4j.DidChangeBuildTarget;
+import ch.epfl.scala.bsp4j.LogMessageParams;
+import ch.epfl.scala.bsp4j.PublishDiagnosticsParams;
+import ch.epfl.scala.bsp4j.ShowMessageParams;
+import ch.epfl.scala.bsp4j.TaskFinishParams;
+import ch.epfl.scala.bsp4j.TaskProgressParams;
+import ch.epfl.scala.bsp4j.TaskStartParams;
+import com.google.common.collect.Sets;
+import io.grpc.ServerBuilder;
+import io.vavr.collection.HashSet;
+import io.vavr.collection.Set;
+import io.vavr.control.Option;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.jetbrains.bsp.bazel.server.bep.BepServer;
+import org.jetbrains.bsp.bazel.server.bsp.info.BspInfo;
+import org.jetbrains.bsp.bazel.server.common.ServerContainer;
+import org.jetbrains.bsp.bazel.server.diagnostics.DiagnosticsService;
+import org.jetbrains.bsp.bazel.server.sync.ProjectStorage;
+import org.jetbrains.bsp.bazel.server.sync.model.Project;
+import org.jetbrains.bsp.bazel.workspacecontext.WorkspaceContextProvider;
+
+class BloopExporter {
+
+  private final BspInfo bspInfo;
+  private final WorkspaceContextProvider workspaceContextProvider;
+  private final Path workspaceRoot;
+
+  public BloopExporter(
+      BspInfo bspInfo, Path workspaceRoot, WorkspaceContextProvider workspaceContextProvider) {
+    this.bspInfo = bspInfo;
+    this.workspaceContextProvider = workspaceContextProvider;
+    this.workspaceRoot = workspaceRoot;
+  }
+
+  public void export() throws BazelExportFailedException {
+    var serverContainer =
+        ServerContainer.create(
+            bspInfo,
+            workspaceContextProvider,
+            Option.of(this.workspaceRoot),
+            Option.of(new NoopProjectStorage()));
+    var projectProvider = serverContainer.projectProvider;
+    var client = new BloopBuildClient(System.out);
+    initializeClient(serverContainer, client);
+
+    Project project = projectProvider.refreshAndGet();
+
+    var failedTargets = client.getFailedTargets();
+    var failedTransitiveTargets =
+        failedTargets.removeAll(
+            project.modules().map(m -> new BuildTargetIdentifier(m.label().getValue())));
+
+    if (failedTransitiveTargets.nonEmpty()) {
```

Comment:
does it mean that if anything in transitive deps fails an entire import fails? we are trying to collect as much data as we can, even some targets are not correct

Reply:
yeah, modules inside the project can fail, but ones outside will fail the import.  I this this is the best compromise:
- stuff being actively developed breaking will still allow you to import
- things outside of the project breaking will fail it, rather than giving bizarre errors at runtime or compile time.